pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -113,6 +114,7 @@ def __init__(
         num_nodes: Optional[int] = None,
         parallel_devices: Optional[List[torch.device]] = None,
         cluster_environment: Optional[ClusterEnvironment] = None,
+        checkpoint_plugin: Optional[CheckpointIOPlugin] = None,
```

Comment:
this doesn't need to be part of the constructor if it raises an exception. we could instead raise an exception if the checkpoint plugin is set in the property setter instead